grass7/raster/r.valley.bottom/r.valley.bottom.py

```diff
@@ -46,196 +46,235 @@
-             e=region.east + (region.ewres * radius))
+    width, height : tuple
+        The width and height of each tile.
+    """
+    n = math.sqrt(n_jobs)
+    width = math.ceil(region.cols / n)
+    height = math.ceil(region.rows / n)
 
-    r.grow(input=input,
-           output=output,
-           radius=radius+1,
-           quiet=True)
+    if width < 250 or height < 250:
+        width = region.cols
+        height = region.rows
 
-    return (region)
+    return width, height
 
 
 def focal_expr(radius, window_square=False):
     """Returns array offsets relative to centre pixel (0,0) for a matrix of
     size radius
 
-    Args
-    ----
+    Parameters
+    ----------
     radius : int
-        Radius of the focal function
-    window_square : bool. Optional (default is False)
-        Boolean to use a circular or square window
+        The radius of the focal function.
+
+    window_square : bool (opt). Default is False
+        Whether to use a circular or square focal window.
 
     Returns
     -------
     offsets : list
         List of pixel positions (row, col) relative to the center pixel
         ( 1, -1)  ( 1, 0)  ( 1, 1)
         ( 0, -1)  ( 0, 0)  ( 0, 1)
-        (-1, -1)  (-1, 0)  (-1, 1)"""
+        (-1, -1)  (-1, 0)  (-1, 1)
+    """
 
     offsets = []
 
     # generate a list of spatial neighbourhood offsets for the chosen radius
     # ignoring the centre cell
     if window_square:
-        
-        for i in range(-radius, radius+1):
-            for j in range(-radius, radius+1):
-                if (i,j) != (0,0):
+        for i in range(-radius, radius + 1):
+            for j in range(-radius, radius + 1):
+                if (i, j) != (0, 0):
                     offsets.append((i, j))
-    
-    else:
 
-        for i in range(-radius, radius+1):
-            for j in range(-radius, radius+1):
+    else:
+        for i in range(-radius, radius + 1):
+            for j in range(-radius, radius + 1):
                 row = i + radius
                 col = j + radius
 
-                if pow(row - radius, 2) + pow(col - radius, 2) <= \
-                    pow(radius, 2) and (i, j) != (0,0):
+                if pow(row - radius, 2) + pow(col - radius, 2) <= pow(radius, 2) and (
+                    i,
+                    j,
+                ) != (0, 0):
                     offsets.append((j, i))
 
     return offsets
 
 
-def get_percentile(L, input, radius=3, window_square=False):
-    """Calculates the percentile whichj is the ratio of the number of points of
-    lower elevation to the total number of points in the surrounding region
+def elevation_percentile(L, input, radius=3, window_square=False, n_jobs=1):
+    """Calculates the percentile which is the ratio of the number of points of lower
+    elevation to the total number of points in the surrounding region
+
+    Notes
+    -----
+    Note this function is currently a bottle-neck is the module because r.mapcalc
```

Comment:
just a small typo: "Note this function is currently a bottle-neck **is** the module" -> "Note this function is currently a bottle-neck **of** the module"